src/modules/ecdh/tests_impl.h

```diff
@@ -142,11 +150,53 @@ static void test_result_basepoint(void) {
     }
 }
 
+static void test_ecdh_wycheproof(void) {
+#include "../../wycheproof/ecdh_secp256k1_test.h"
+    int t;
+    for (t = 0; t < SECP256K1_ECDH_WYCHEPROOF_NUMBER_TESTVECTORS; t++) {
+        int parsed_ok;
+        secp256k1_pubkey point;
+        const unsigned char *pk;
+        const unsigned char *sk;
+        const unsigned char *expected_shared_secret;
+        unsigned char output_ecdh[65] = { 0 };
+
+        int expected_result;
+        int actual;
+
+        memset(&point, 0, sizeof(point));
+        pk = &wycheproof_ecdh_public_keys[testvectors[t].pk_offset];
+        parsed_ok = secp256k1_ec_pubkey_parse(CTX, &point, pk, testvectors[t].pk_len);
+
+        expected_result = testvectors[t].expected_result;
+
+        /* fail if public key is valid, but doesn't parse */
+        CHECK(parsed_ok || expected_result == 0);
```

Comment:
I think this should be tighter:
```suggestion
        CHECK(parsed_ok == expected_result);
```

The missing case is `parsed_ok == 1` and `expected_result == 0`. If I'm not mistaken, then the entire test doesn't fail currently:
 - the CHECK here passes
 - we don't `continue`
 - but then the `CHECK(secp256k1_memcmp_var`)...` passes vacuously because `testvectors[t].shared_len == 0`